### ISSUE TITLE
Need to delete a particular pattern of urls which were indexed and pushed to Google Cloud in past .

### ISSUE THREAD

**Issue by sudeshna-majumder** [Norconex/collector-http#706]
I need to exclude few urls patterns from getting indexed , which is happening with exclude pattern but now I also want to delete all similar patterned urls which were indexed and pushed to Google Cloud in past . How can I do that ? Need help. The domain for which I need to delete the urls are being crawled under default crawler .

**Comment by sudeshna-majumder** [comment]
I need to exclude few urls patterns from getting indexed , which is happening with exclude pattern but now I also want to delete all similar patterned urls which were indexed and pushed to Google Cloud in past . How can I do that ? Need help. The domain for which I need to delete the urls are being crawled under default crawler .

**Comment by sudeshna-majumder** [comment]
I have added regex to my referenceFilter with onMatch EXCLUDE , which is working , it's excluding the pattern as per log . But is spite of adding <orphansStrategy> the same patterned urls which were indexed previously are not getting deleted . No .del.ref files created for them. How can I get those urls deleted.

**Comment by essiembre** [comment]
It is possible after a few runs that the collector has lost knowledge of your excluded URLs.  Did you have the orphan strategy to DELETE at the time you added the exclusion?   In any case, if it ran a few times since, it is likely too late to go back have those processed as deletions.  In which case, it is probably easier to simply delete those documents directly from Google Cloud Search.

Another approach is to also index the crawl date. That way you can delete documents crawled before X date.  You can use the `CurrentDateTagger` from the Importer module to that effect.

**Comment by sudeshna-majumder** [comment]
Hi Pascal,

I am attaching my config file and logs(not the complete crawler log , but hope will be useful to find required info) here.
In the config it is referring to another file for 'exclude pattern regex' . Pattern can found in the log(ex. https://sitename.com/fr/node/641). I need my crawler to send delete request to committer for those urls as well along with rejecting them. But those are not deleted . Even after several runs of changing the configs similar indexes are still present and urls are searchable.
I have tried rerunning multiple times by clearing queue and crawlstore and also few times by clearing only queue not the crwalstore . But seems nothing works. Can you please help?
 Also is there any way to send delete request to committer for those urls by making manual entries to del.ref file or something similar ?

Thanks,
Sudeshna

[crawlerdefault.txt](https://github.com/Norconex/collector-http/files/4910001/crawlerdefault.txt)
[crawlerdefult.txt](https://github.com/Norconex/collector-http/files/4910003/crawlerdefult.txt)

**Comment by sudeshna-majumder** [comment]
Hi Pascal,

Thanks for your reply. 

No , i didn't have orphansStrategy as DELETE while changing config. 

Curious about the another approach . If CurrentDateTagger was not in place when those URLs got indexed will I be able to reach them with this approach ?

And also if I am able to list the URLs to be deleted is there any way to create and send delete request to committer for them ?

Thanks,
Sudeshna

**Comment by essiembre** [comment]
Your original orphan strategy explains it then.  

Unfortunately, the committer does not include/duplicate the general maintenance/management capabilities of your search engine.  You typically would have to perform those one-time cleanup operations against your search engine directly.  So in your case, you could send a manual deletion request for all these URLs you have identified.

If instead, you want to rely on the crawl date trick, here is one way to do it:

1. Add the crawl date to your config/search engine.
2. Perform a full crawl (remove the "workdir" so it crawls as if the first time)
3. Once crawling is done, delete all records with an empty crawl date field.

In the future, you would instead delete anything older than X.

Doe this work for you?

**Comment by sudeshna-majumder** [comment]
Thanks Pascal.

Seems like would serve my purpose .I'll give it a try .

**Comment by sudeshna-majumder** [comment]
Hi Pascal,

Thanks a ton . I am finally able to delete those documents. And I did it through crawler not my manually cleaning my indexes.

And you were absolutely right about the behavior of  'orphasStrategy' . As 'orphansStrategy' wasn't set as 'DELETE' when changing referenceFiler regex , after few runs crawler lost knowledge of previously indexed docs. 

I did a fresh crawl by removing referenceFilter and then I added my referenceFilter along with orphansStrategy 'DELETE', which deleted all those patterned URLs along with older docs .
Also added CurrentDateTagger for  future.
